misk-crypto/src/main/kotlin/misk/crypto/pgp/RealPgpDecrypter.kt

```diff
@@ -12,6 +12,14 @@ import org.bouncycastle.openpgp.jcajce.JcaPGPObjectFactory
 import org.bouncycastle.openpgp.operator.jcajce.JcePublicKeyDataDecryptorFactoryBuilder
 import org.bouncycastle.util.io.Streams
 
+/**
+ * RealPgpDecrypter is a simple PGP decryption helper.
+ * The implementation makes a  couple key assumptions:
+ * - The input data is public key encrypted.
+ * - The input data contains only ONE encrypted payload.
+ * - The first OR second packet in the input must be the encryption method packet.
```

Comment:
Is this any different than "the input data is PGP encrypted"? I would imagine that this is literally required by the format

Reply:
I mean, it's dumb, but the spec doesn't point it out. Depending on the PGP client used to generate the payload, you could have none encryption packets before the actual data. In that case, instead of ignoring them, we'd fail ungracefully (since the cast would fail)